src/components/donation/Steps/DonateMethods/Stocks/Form.tsx

```diff
@@ -31,7 +31,10 @@ export default function Form(props: StockFormStep) {
     resolver: yupResolver(
       schema<FV>({
         symbol: requiredString.trim(),
-        numShares: requiredString.trim().matches(/^[1-9]\d*$/, "invalid"),
+        numShares: stringNumber(
+          (s) => s.required("required"),
```

Comment:
more recent `stringNumber` is more accurate/descriptive than old regex validator. this however necessitates changing `number` to `string`